include/fmt/format.h

```diff
@@ -3198,17 +3198,24 @@ FMT_CONSTEXPR basic_string_view<Char> compile_string_to_view(
   return {s.data(), s.size()};
 }
 
-#define FMT_STRING_IMPL(s, base)                                  \
-  [] {                                                            \
-    /* Use a macro-like name to avoid shadowing warnings. */      \
-    struct FMT_COMPILE_STRING : base {                            \
-      using char_type = fmt::remove_cvref_t<decltype(s[0])>;      \
-      FMT_MAYBE_UNUSED FMT_CONSTEXPR                              \
-      operator fmt::basic_string_view<char_type>() const {        \
-        return fmt::detail::compile_string_to_view<char_type>(s); \
-      }                                                           \
-    };                                                            \
-    return FMT_COMPILE_STRING();                                  \
+#if FMT_GCC_VERSION
+#  define FMT_COMPILE_STRING_VISIBILITY_ATTRIBUTE \
```

Comment:
I suggest renaming `FMT_COMPILE_STRING_VISIBILITY_ATTRIBUTE ` to `FMT_GCC_VISIBILITY_HIDDEN` to indicate that it's a gcc attribute and that it can be applied in other places (not specific to COMPILE_STRING).

Reply:
Should it be placed in some other place then?

Reply:
Sure, somewhere next to https://github.com/fmtlib/fmt/blob/cba5970cd8a330b5f4d4a26281121fac90815a6b/include/fmt/format.h#L66-L70

Reply:
ok, and another question: since it is not only GCC-specific maybe the name should be `FMT_VISIBILITY_HIDDEN`?

Reply:
don't mind, it's probably out of the scope of this PR, so I will create only GCC-specific `FMT_GCC_VISIBILITY_HIDDEN`
IMHO, if there will be a need for the same attribute but for all compilers then `FMT_VISIBILITY_HIDDEN` should be added with `FMT_GCC_VISIBILITY_HIDDEN` inside GCC-specific branch

Reply:
renamed to `FMT_GCC_VISIBILITY_HIDDEN` and moved